SideBar.py

```
@@ -90,7 +90,73 @@ def copy(self, source, destination):
 			shutil.copy2(source, destination)
 
 		if self.view:
-			self.view.erase_status('ZZZ')
+			self.view.erase_status('SideBarTools:Copy')
 
 	def description(self):
 		return 'Duplicate File…'
+
+
+def temporary_status_message(view, message, key='SideBarTools', duration=5000):
+	view.set_status(key, message)
+
+	def erase_status():
+		view.erase_status(key)
+
+	sublime.set_timeout_async(erase_status, duration)
+
+
+class SideBarMoveCommand(SideBarCommand):
+
+	def run(self, paths):
+		self.view = self.window.active_view()
+		self.source = self.get_path(paths)
+
+		input_panel = self.window.show_input_panel(
+			'Move to:', self.source, self.on_done, None, None)
+
+		base, leaf = os.path.split(self.source)
+		name, ext = os.path.splitext(leaf)
+
+		input_panel.sel().clear()
+		input_panel.sel().add(sublime.Region(len(base) + 1, len(self.source) - len(ext)))
+
+	def on_done(self, destination):
+		threading.Thread(target=self.move, args=(self.source, destination)).start()
+
+	def move(self, source, destination):
+		print(source, destination)
+		if self.view:
+			self.view.set_status(
+				'SideBarTools:Move', 'Moving "{}" to "{}"'.format(source, destination))
+		else:
+			self.window.status_message(
+				'Moving "{}" to "{}"'.format(source, destination))
+
+		destination_dir = os.path.dirname(destination)
+		try:
+			os.makedirs(destination_dir)
+		except OSError:
+			print('Destination directory seems to exists...')
```

Having this feedback in the console seems unnecessary to me. There is nothing wrong with the destination already existing.

---

You're right, actually I was more concerned about the case when the path exists but Sublime can't write to it, but even so, there's nothing we can do from within the plugin,

---

Ah, I was just getting this for existing paths that were no problem (which is pretty much every time I moved something). It *would* be super useful if we get a message if Sublime can't write to the target.. doesn't the other error message already catch this? (haven't figured out how to test this case).

---

Well there's an easy way of testing this:
- Create the test folder
- Change permissions so your current user can't write to it: assuming you use macOS: something along the lines of `chmod a-rwx,u+rwX,a+rX test-folder/` then `chown root:root test-folder/`; on POSIX platforms, those commands should make that folder only writeable by root, readable by everyone else.
- Try to move a file with the plugin, it should fail and you should get an exception or something in ST's console.

To alert the user about this error, we could use an actual message box, instead of status messages. This error is serious enough (should be rare enough) to force the user to dismiss the message and fix the permissions issue by other means.

---

>To alert the user about this error, we could use an actual message box, instead of status messages.

I think if the file doesn't move that's already feedback, plus something in the statusbar. I'd reserve message boxes for if the program gets in a state that needs user intervention, or that could have caused data-loss. If the move didn't happen, there isn't anything wrong yet.

---

> Change permissions so your current user can't write to it: assuming you use macOS: something along the lines of chmod a-rwx,u+rwX,a+rX test-folder/ then chown root:root test-folder/;.

Doh... I knew that. How many days till weekend?

Anyway, works just fine. I think it's good without extra messaging. However, I do think the message tends to get too long. I would propose formatting it like this:

`Error moving: {error} ({src} to {dst})` 

This results in the reason (e.g. "permission denied") leading the message and makes it more likely for the important part to fit in the statusbar.

---

Well... I mean, the expectation is that copying/moving just works, but this is not really a file manager, which the main interface is a list of file/folders. The only feedback we have that the operation succeeded is maybe a quick sidebar refresh, and a status bar message. But the sidebar is all the way to the left and the status bar is all the way to the bottom, far from the main application interface, which is a text buffer with pretty colors. And with the current high-res displays, they are further away from the center. If the operation fails, we only get a small text in the status bar, and current themes' styling doesn't help to draw attention to it.

---

I guess I'm on my 1280x800 (@2x though) MacBook too much 😄 . You do have to interact with the panel all the way at the bottom to enact the move, so possibly your eyes are already paying attention to the important parts of the UI. Depending on your attention span 😉 

Duplicate can also attempt to create a new file in a non-accessible directory, as does the default rename command. Both fail silently. I think it's easy enough to verify if the desired effect occurred, and there is nothing wrong if it didn't. Having any feedback at all in the statusbar seems like good thing, and perhaps we should add it for duplicate, but let's not overdo it. I understand this is for a big part also my personal preference to stay away from the message boxes unless something pretty severe happened.

---

I just added a `try..except` handler for Duplicate, now both operations use `Window.status_message` if there's an error.

Yes, everyone's setup is different. In any case, we can keep using the status bar for errors like this, then revisit this decision later, or if someone else asks :)